pkg/daemon/update.go

```diff
@@ -556,6 +612,122 @@ func (dn *Daemon) updateKernelArguments(oldConfig, newConfig *mcfgv1.MachineConf
 	return exec.Command("rpm-ostree", args...).Run()
 }
 
+// MountOSContainer mounts the container and returns the mountpoint
+func (dn *Daemon) MountOSContainer(container string) (mnt, containerName string, err error) {
+	var authArgs []string
+	if _, err = os.Stat(kubeletAuthFile); err == nil {
+		authArgs = append(authArgs, "--authfile", kubeletAuthFile)
+	}
+	// Pull the image
+	args := []string{"pull", "-q"}
+	args = append(args, authArgs...)
+	args = append(args, container)
+	pivotutils.RunExt(false, numRetriesNetCommands, "podman", args...)
+
+	containerName = "mcd-" + string(uuid.NewUUID())
+	// `podman mount` wants a container, so let's create a dummy one, but not run it
+	var cidBuf []byte
+	cidBuf, err = runGetOut("podman", "create", "--net=none", "--annotation=org.openshift.machineconfigoperator.pivot=true", "--name", containerName, container)
+	if err != nil {
+		return
+	}
+
+	cid := strings.TrimSpace(string(cidBuf))
+	// Use the container ID to find its mount point
+	mntBuf, err := runGetOut("podman", "mount", cid)
+	if err != nil {
+		return
+	}
+	mnt = strings.TrimSpace(string(mntBuf))
+	return
+}
+
+// switchKernel updates kernel on host with the kernelType specified in MachineConfig.
+// Right now it supports default (traditional) and realtime kernel
+func (dn *Daemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error {
+	// We support Kernel update only on RHCOS nodes
```

Comment:
should this function check if we need to do all this?

Reply:
```
I0122 13:16:51.153341    2609 update.go:1222] Initiating switch from kernel default to default
I0122 13:16:51.159415    2609 run.go:16] Running: podman pull -q --authfile /var/lib/kubelet/config.json registry.svc.ci.openshift.org/ci-op-6mmddzs0/stable@sha256:b2e468365bbc6808c581ee7a7d452bb35c63e1fac05c1501621a76ccb4f8b276
151450e66d9aab561ddc0e94804d9f7b46f1f987f3c9e439f5b746f6a03c2152
I0122 13:17:28.373015    2609 rpm-ostree.go:366] Running captured: podman create --net=none --annotation=org.openshift.machineconfigoperator.pivot=true --name mcd-75d0c6bc-fed2-43f4-8ee5-42c09322fcab registry.svc.ci.openshift.org/ci-op-6mmddzs0/stable@sha256:b2e468365bbc6808c581ee7a7d452bb35c63e1fac05c1501621a76ccb4f8b276
I0122 13:17:28.645906    2609 rpm-ostree.go:366] Running captured: podman mount c6dde396d617bd3e6f74a916b8c753530f7b17c768ac7e7436eb18781e417107
I0122 13:17:28.884382    2609 update.go:1222] Switching to kernelType=default,invoking rpm-ostree ["override" "remove" "kernel" "kernel-core" "kernel-modules" "kernel-modules-extra" "--install" "/var/lib/containers/storage/overlay/b4d98bd61ab6ca1e038351946499de5d802735f014712c4f379875c1856822a0/merged/kernel-rt-core-4.18.0-147.3.1.rt24.96.el8_1.x86_64.rpm" "--install" "/var/lib/containers/storage/overlay/b4d98bd61ab6ca1e038351946499de5d802735f014712c4f379875c1856822a0/merged/kernel-rt-modules-4.18.0-147.3.1.rt24.96.el8_1.x86_64.rpm" "--install" "/var/lib/containers/storage/overlay/b4d98bd61ab6ca1e038351946499de5d802735f014712c4f379875c1856822a0/merged/kernel-rt-modules-extra-4.18.0-147.3.1.rt24.96.el8_1.x86_64.rpm"]
```

this is what the MCD says, basically we don't need to do this _every time_ we need to check if old->new config is asking for a switch, otherwise do as it was nothing

Reply:
yeah right, it should do nothing when kernelType is default and it hasn't been changed.

Reply:
Thanks for catching this! 
Should have looked earlier into the journal logs of this PR. It wasn't obvious because of the similar error message we were seeing in the gcp-op CI in other PRs.